grasa_event_locator/views.py

```diff
@@ -49,6 +49,10 @@ def admin_user(request):
         newUser.save()
         uInfo = userInfo(user=newUser, org_name="Administrator", isAdmin=True, isPending=False)
         uInfo.save()
+        if int(uInfo.id) != int(uInfo.user_id):
```

Comment:
It seems like doing these checks might not be necessary, it's always supposed to match up so you could just set the first instance "uInfo" to have pk=int(newUser.id) right?

Reply:
Yup, while the numbers are supposed to line up in typical use, we've given them some pretty obvious ways to screw with the numbering.

For instance, going into mySQL and messing with the primary key numbers, which will then botch up the next entry in the table, specifically making the next entry in the table's id be off compared to the number of users. This patch would notice that, and sort out the next entry's ID to match the actual user it's associated with.

Something else that could happen is that someone runs createsuperuser. This advances the count of the number of users by 1, but does not create a new table entry, which would then throw off the next entry submitted via the registration form; it's PK would be one less than the user_id it's associated with.

While the two scenarios above are unlikely to happen in regular use, I feel it would be best to at least prepare for the possibility of it happening, since anyone with knowledge of the terminal (certainly Monroe County IT) could perform the above. Plus if the ID numbers do screw up through unknown means, then at least the system will still be able to register new users while the admins rummage through the database to fix things up.

Reply:
The only situation this fails in is if the user_id equals an existing primary key (perhaps someone changed another entry's primary key to an existing user's id). But at that point, that's the IT department's business to sort out.

Reply:
Oh, regarding the point of doing this during the Admin creation. If someone runs createsuperuser first before clicking the header button, it'll mess up the number of the table. The check here is in case that happens; the admin created by the header button will still be able to log in.